api/ocm/plugin/README.md

```diff
@@ -0,0 +1,151 @@
+# The OCM Library Plugin Concept
+
+The ocm library supports a plugin mechanism to provide further variants
+for OCM and library extension points without the need of extending and recompiling
+the OCM CLI (and other applications using this library).
+
+This plugin concept is not part of the OCM specification, because it is
+a feature of this library implementation.
+
+The following extension points are supported:
+
+- Access methods
+- Uploaders
+- Downloaders
+- Actions
+- Value sets (for example for routing slip entries)
+- Config types
+- Value Merge Handler (for example for label values in delta transports)
+- CLI Commands (for OCM CLI)
```

Comment:
I dont understand why we can add CLI Commands to a plugin infrastructure that is supposedly an interface of the library. To me that means the plugin system is not confined to the library, but to the entire codebase.
For me that means either that
- the whole plugin system is across the entirety of the OCM codebase
- we have 2 separate plugin systems that need to be documented separately and be detached from each other